src/main/java/uk/nhs/hee/tis/trainee/sync/event/ProgrammeListener.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2024 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.sync.event;
+
+import static io.awspring.cloud.messaging.listener.SqsMessageDeletionPolicy.ON_SUCCESS;
+
+import io.awspring.cloud.messaging.listener.annotation.SqsListener;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.nhs.hee.tis.trainee.sync.model.Programme;
+import uk.nhs.hee.tis.trainee.sync.service.ProgrammeSyncService;
+
+/**
+ * A listener for Programme messages.
+ */
+@Slf4j
+@Component
+public class ProgrammeListener {
+
+  private final ProgrammeSyncService programmeSyncService;
+
+  ProgrammeListener(ProgrammeSyncService programmeSyncService) {
+    this.programmeSyncService = programmeSyncService;
+  }
+
+  @SqsListener(value = "${application.aws.sqs.programme}", deletionPolicy = ON_SUCCESS)
+  void getProgramme(Programme programme) {
+    log.debug("Received programme {}.", programme);
+    //Once RO data will be added, then call programmeSyncService.syncRecord(programme)
```

Comment:
At this point, we simply ignore the message, because we don't yet have the RO information (it is in the HeeUser / UserRole / UserDesignatedBodyCode data which is not yet included). That enrichment of the related programme memberships will be handled in a subsequent PR.